STEER/STEER/AliGenerator.cxx

```diff
@@ -450,19 +450,33 @@ void AliGenerator:: SetHighWaterMark(Int_t nt)
   //
   if (fStack)
      fStack->SetHighWaterMark(nt);
-  else 
+  else
      gAlice->GetMCApp()->SetHighWaterMark(nt);
-   
+
 }
 void AliGenerator::FinishRun()
 {
     ;
 }
 
+///-------------------------------------------------------------------------
+Bool_t AliGenerator:: ApplyUserTrigger()
```

Comment:
@lucamicheletti93 : please check teh return value, in some branches of your conditions, you do not return anything.

Reply:
I think the issue is due to that if ApplyUserTrigger() is called but fSetUserTrig == kFALSE nothing is returned. I should add a return kTRUE in case fSetUserTrig == kFALSE.

Reply:
Yes. You should be sure that something is always returned.

Reply:
I re-commit the code after applying this correction!